packages/xo-server/src/api/pool.mjs

```diff
@@ -1,11 +1,73 @@
 import { asyncMap } from '@xen-orchestra/async-map'
+import { createLogger } from '@xen-orchestra/log'
 import { defer as deferrable } from 'golike-defer'
 import { format } from 'json-rpc-peer'
 import { Ref } from 'xen-api'
-import { incorrectState } from 'xo-common/api-errors.js'
+import { forbiddenOperation, incorrectState } from 'xo-common/api-errors.js'
 
 import { moveFirst } from '../_moveFirst.mjs'
 
+const log = createLogger('xo:pool')
+
+// ===================================================================
+
+export async function backupGuard(poolId) {
+  const jobs = await this.getAllJobs()
+  const guard = id => {
+    if (this.getObject(id).$poolId === poolId) {
+      throw forbiddenOperation('Rolling pool update', `A backup is running on the pool: ${poolId}`)
+    }
+  }
+
+  jobs.forEach(({ runId, type, vms }) => {
+    if (runId !== undefined && type === 'backup') {
+      if (vms.id !== undefined) {
+        const id = vms.id.__or ?? vms.id
+        if (Array.isArray(id)) {
+          id.forEach(guard)
+        } else {
+          guard(id)
+        }
+      } else {
+        /**
+         * Smart mode
+         * vms: {
+         *    $pool: {
+         *      __and: [
+         *        {
+         *          __not: {
+         *            __or: ['poolId','poolId']
+         *          }
+         *        },
+         *        {
+         *          __or: ['poolId','poolId']
+         *        }
+         *      ]
+         *    }
+         *  }
+         */
+        let isPoolSafe = false
+        vms.$pool.__and.forEach(conditions => {
```

Comment:
would it possible to use an already defined method, like the value-matcher package ?